api/src/components/flag/controller.ts

```diff
@@ -53,13 +53,23 @@ export const createFlag = async (
     try {
         const publication = await publicationService.get(event.pathParameters.id);
 
-        if (!publication || publication.currentStatus !== 'LIVE') {
+        if (!publication || !publication.versions.some((version) => version.currentStatus === 'LIVE')) {
             return response.json(404, {
-                message: 'Cannot flag that a publication that does not exist, or is not LIVE'
+                message: 'Cannot flag that a publication that does not exist, or has never been LIVE'
             });
         }
 
-        if (publication.user.id === event.user.id) {
+        // Latest published version (the version presented when someone tries to raise a flag) is either current version,
+        // or if that's not LIVE, the version before, as we've already confirmed that the publication has been live at some point.
+        const latestPublishedVersion =
+            publication.versions.find((version) => version.isCurrent && version.currentStatus === 'LIVE') ||
+            publication.versions.find((version) => version.versionNumber === publication.versions.length - 1);
```

Comment:
Is the second condition (after || ) really needed? It allows users to flag DRAFT versions.